CMakeLists.txt

```diff
@@ -1,111 +1,104 @@
-cmake_minimum_required(VERSION 3.14)
+cmake_minimum_required(VERSION 3.14...3.20)
 
 project(
-    example
-    VERSION 0.1.0
-    LANGUAGES CXX
+  example
+  VERSION 0.1.0
+  LANGUAGES CXX
 )
 
+option(BUILD_SHARED_LIBS "default" YES)
+
 # ---- Warning guard ----
 
 # Protect dependents from this project's warnings if the guard isn't disabled
-set(example_warning_guard SYSTEM)
-if(example_INCLUDE_WITHOUT_SYSTEM)
-  set(example_warning_guard "")
-endif()
+## XXX why? CK
+## set(example_warning_guard SYSTEM)
+## if(example_INCLUDE_WITHOUT_SYSTEM)
+##   set(example_warning_guard "")
+## endif()
 
 # ---- Declare library ----
 
 include(GenerateExportHeader)
 
-set(headers include/example/main.h)
+include(GNUInstallDirs)
+set(EXAMPLE_INCLUDE_DIRECTORY "${CMAKE_INSTALL_INCLUDEDIR}")
+
+set(EXAMPLE_headers include/example/main.h)
 
-set(sources source/main.cpp)
+set(EXAMPLE_sources source/main.cpp)
 
-add_library(example_example ${sources} ${headers})
+add_library(example_example ${EXAMPLE_sources} ${EXAMPLE_headers})
 add_library(example::example ALIAS example_example)
 
 generate_export_header(
-    example_example
-    BASE_NAME example
-    EXPORT_FILE_NAME "include/example/example_export.h"
+  example_example BASE_NAME example EXPORT_FILE_NAME
+  "include/example/example_export.h"
 )
 
 set_target_properties(
-    example_example PROPERTIES
-    CXX_VISIBILITY_PRESET hidden
-    VISIBILITY_INLINES_HIDDEN YES
-    VERSION "${PROJECT_VERSION}"
-    SOVERSION "${PROJECT_VERSION_MAJOR}"
-    EXPORT_NAME example
-    OUTPUT_NAME example
+  example_example
+  PROPERTIES CXX_VISIBILITY_PRESET hidden
+             VISIBILITY_INLINES_HIDDEN YES
+             VERSION "${PROJECT_VERSION}"
+             SOVERSION "${PROJECT_VERSION_MAJOR}"
+             EXPORT_NAME example
+             OUTPUT_NAME example
```

Comment:
This is a bogus formatting style, I will never understand support this insanity.